src/python/pants/bin/pants_runner.py

```diff
@@ -80,6 +83,17 @@ def run(self, start_time: float) -> ExitCode:
         ):
             # N.B. We inline imports to speed up the python thin client run, and avoids importing
             # engine types until after the runner has had a chance to set __PANTS_BIN_NAME.
+            if "SCIE" not in os.environ and "NO_SCIE_WARNING" not in os.environ:
+                warn_or_error(
+                    "2.17.0.dev5",
```

Comment:
This can't be right. Did you mean at least 2.18?

Reply:
Thanks. It seems to me cherry-pick retro-deprecations should be outright ~outlawed, certainly once the alpha corner has turned to rc. It's a strange universe where the stabilization phase introduces a deprecation - it smacks of poor planning / impulsiveness in general and I associate dev -> a -> rc -> stable with some sort of release process advertising itself as on the conservative end of the spectrum.

Reply:
the PR classification answers this without doubt: "user api change" should as a rule never be picked. (we have exceptions where the change is more of a fix in nature; as we've seen recently)

Reply:
You're right, it was over-eagerness on my part. Especially since non-trivial numbers of users do use rcs.